arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -62,6 +62,29 @@ config MPU_ALLOW_FLASH_WRITE
 	help
 	  Enable this to allow MPU RWX access to flash memory
 
+config MPU_SRAM_GAP_FILLING
```

Comment:
>Why do we need this?
All the gap filling does is restrict access to supervisor mode threads, which we don't need for user mode.
I would rather that this option, when not enabled, does no automatic gap filling at all.

We do not need this, @andrewboie, _right now_ because any static regions we enforce at boot time:
- code (mandatory)
- execute-from-sram (optional)
- no-cache (optional)
- read-only (optional)
- gcov (optional)
- ....
are properly partitioning the memory map, i.e. they do not leave gaps in between them. So, not allowing gaps does not make us waste regions. Therefore, the MPU driver behavior when programming the static regions is the same, regardless of whether MPU_SRAM_GAP_FILLING is set or unset.

In other words, we only need (_and should allow_) to have gaps in the sram part where we have dynamic regions defined (i.e. the kernel sram area). So I wanted to restrict the Kconfig option, to reflect _exactly_ this. I don't see why we need the Kconfig option to mean something more generic; I have no plans to allow the MPU driver to skip gap filling outside the kernel SRAM, anyways.

Do you anticipate the need (in the near future) of a Zephyr memory map organization (done at linker level), which will leave gaps between the static sections?

Reply:
> And what do you propose for the Kconfig option ?

CONFIG_MPU_GAP_FILLING?

Reply:
@andrewboie that's fine with me, obviously
But do you approve the help text below the option? Also copying it here for reference:

```
          This Kconfig option instructs the MPU driver to enforce
	  a full kernel SRAM partitioning, when it programs the
	  dynamic MPU regions (user thread stack, PRIV stack guard
	  and application memory domains) during context-switch. We
	  allow this to be a configurable option, in order to be able
	  to switch the option off and have an increased number of MPU
	  regions available for application memory domain programming.
	  Note that an increased number of MPU regions should only be
	  required when building with USERSPACE support.

```

Reply:
> CONFIG_MPU_GAP_FILLING

Fine with me (just have to remove SRAM from the name)

Reply:
Done now @andrewboie